### PR TITLE
MM-38150 System console additions for playbook permissions

### PR DESCRIPTION
**Pull request by crspeller**
#### Summary
Complement to server pr https://github.com/mattermost/mattermost-server/pull/19096 see the full description there. 

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-38504

#### Screenshots
image

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Review by cpoile — APPROVED** [review]

File: components/admin_console/permission_schemes_settings/permission_team_scheme_settings/permission_team_scheme_settings.tsx [hunk 1]
```diff
@@ -430,7 +484,7 @@ export default class PermissionTeamSchemeSettings extends React.PureComponent<Pr
         this.props.history.push('/admin_console/user_management/permissions');
     }
 
-    toggleRole = (roleId: 'all_users' | 'team_admin' | 'channel_admin' | 'guests') => {
+    toggleRole = (roleId: 'all_users' | 'team_admin' | 'channel_admin' | 'guests' | 'playbook_admin') => {
```
Comment (mkraft) [inline components/admin_console/permission_schemes_settings/permission_team_scheme_settings/permission_team_scheme_settings.tsx:487]:
```suggestion
    toggleRole = (roleId: 'all_users' | 'team_admin' | 'channel_admin' | 'guests' | GeneralConstants.PLAYBOOK_ADMIN_ROLE) => {
```

File: components/admin_console/permission_schemes_settings/permission_team_scheme_settings/permission_team_scheme_settings.tsx [hunk 2]
```diff
@@ -690,6 +748,26 @@ export default class PermissionTeamSchemeSettings extends React.PureComponent<Pr
                             />
                         </AdminPanelTogglable>
 
+                        <AdminPanelTogglable
+                            className='permissions-block'
+                            open={this.state.openRoles.playbook_admin}
+                            onToggle={() => this.toggleRole('playbook_admin')}
```
Comment (mkraft) [inline components/admin_console/permission_schemes_settings/permission_team_scheme_settings/permission_team_scheme_settings.tsx:754]:
```suggestion
                            onToggle={() => this.toggleRole(GeneralConstants.PLAYBOOK_ADMIN_ROLE)}
```

File: components/admin_console/permission_schemes_settings/permission_team_scheme_settings/permission_team_scheme_settings.tsx [hunk 3]
```diff
@@ -204,6 +220,9 @@ export default class PermissionTeamSchemeSettings extends React.PureComponent<Pr
                 channelGuest = this.props.roles[this.props.scheme.default_channel_guest_role];
                 channelUser = this.props.roles[this.props.scheme.default_channel_user_role];
                 channelAdmin = this.props.roles[this.props.scheme.default_channel_admin_role];
+                playbookAdmin = this.props.roles[this.props.scheme.default_playbook_admin_role];
+                playbookMember = this.props.roles[this.props.scheme.default_playbook_member_role];
+                runMember = this.props.roles[this.props.scheme.default_run_member_role];
```
[the remark this answers is not in the document]
Comment (crspeller) [reply components/admin_console/permission_schemes_settings/permission_team_scheme_settings/permission_team_scheme_settings.tsx:225]: Correct. There aren't actually any user changeable permissions associated with it yet.

**Comment by crspeller** [comment]
@mkraft Fixed the spelling. Going to do the constant changes as a separate PR as they are not limited to my changes.

**Pull request merged**